### PR TITLE
added enum to `message.type`... probably

### PR DESCRIPTION
**Pull request by namanyt**
Not sure if the enum thinks works 100%... but it passed all tests

### CODE REVIEW

**Comment by namanyt** [comment]
#10

File: chatto/message.py [hunk 1]
```diff
@@ -37,22 +38,48 @@
...
 class Message:
     id: str
-    type: str
+    type: MessageTypes
     stream: Stream
     channel: Channel
     published_at: dt.datetime
     content: str
 
+    def get_type(type: str) -> MessageTypes:
```
Comment (namanyt) [inline chatto/message.py:63]: make this a function cus yt api returns a str for the api type... this func is converting it to the enum

File: chatto/message.py [hunk 2]
```diff
@@ -37,22 +38,48 @@
...
+    MessageDeletedEvent = "messageDeletedEvent" 
+    NewSponsorEvent = "newSponsorEvent"
+    SponsorOnlyModeEndedEvent = "sponsorOnlyModeEndedEvent"
+    SponsorOnlyModeStartedEvent = "sponsorOnlyModeStartedEvent"
+    MemberMilestoneChatEvent = "memberMilestoneChatEvent"
+    SuperChatEvent = "superChatEvent"
+    SuperStickerEvent = "superStickerEvent"
+    TextMessageEvent = "textMessageEvent"
+    Tombstone = "tombstone"
+    UserBannedEvent = "userBannedEvent"
```
Comment (parafoxia) [inline chatto/message.py:52]: Enum keys should be upper-snake-case. For example, line 42 would look like:
```py
CHAT_ENDED_EVENT = "chatEndedEvent"
```
Remember to make this change elsewhere in the code too.

File: chatto/message.py [hunk 3]
```diff
@@ -37,22 +38,48 @@
 from chatto.channel import Channel
 from chatto.stream import Stream
 
+class MessageTypes(enum.Enum):
```
Comment (parafoxia) [inline chatto/message.py:41]:
```suggestion
class MessageType(enum.Enum):
```

File: chatto/message.py [hunk 4]
```diff
@@ -37,22 +38,48 @@
...
+    SuperStickerEvent = "superStickerEvent"
+    TextMessageEvent = "textMessageEvent"
+    Tombstone = "tombstone"
+    UserBannedEvent = "userBannedEvent"
 
 @dataclass(eq=True, frozen=True)
 class Message:
     id: str
-    type: str
+    type: MessageTypes
```
Comment (parafoxia) [inline chatto/message.py:57]:
```suggestion
    type: MessageType
```
Remember to make this change elsewhere in the code too.

File: chatto/message.py [hunk 5]
```diff
@@ -37,22 +38,48 @@
...
+        if (type == MessageTypes.Tombstone.value): return MessageTypes.Tombstone
+        if (type == MessageTypes.UserBannedEvent.value): return MessageTypes.UserBannedEvent
+
     @classmethod
     def from_youtube(cls, item: dict[str, t.Any], stream: Stream) -> Message:
         snippet = item["snippet"]
         return cls(
             id=item["id"],
-            type=snippet["type"],
+            type=cls.get_type(snippet["type"]),
```
Comment (parafoxia) [inline chatto/message.py:82]:
```suggestion
            type=MessageType(snippet["type"]),
```

File: tests/test_message.py [hunk 6]
```diff
@@ -35,7 +35,7 @@
 from dateutil.tz import tzutc
 
 from chatto.channel import Channel
-from chatto.message import Message
+from chatto.message import Message, MessageTypes
```
Comment (parafoxia) [inline tests/test_message.py:38]:
```suggestion
from chatto.message import Message, MessageType
```
Make the changes above in this file too.

**Review by parafoxia — CHANGES_REQUESTED** [review]
Thanks for taking the time to implement this! Saves me a good bit of looking up.

Just a few small changes related to name changes and ways of doing certain things, but good overall.

Also make sure all nox tests pass before committing. You can run `nox` to run them all.

File: chatto/message.py (hunk 2 above)
Comment (namanyt) [reply chatto/message.py:52]: on it

**Review by parafoxia — APPROVED** [review]
Looks good to me! I'll leave this unmerged for now as there's other things I want to release first, but I'll double check it works fine (it looks like it should) and then release it into v0.5.

Thanks again!

**Comment by parafoxia** [comment]
Sorry about this, but the changes I wanted to make for v0.4.2 are creating too many merge conflicts with this PR to merge. I tried to fix it but I've just made it worse 😅 Can you open a new PR re-adding these changes to the latest commit of the main branch? Thanks!

**Review by parafoxia — APPROVED** [review]
This looks like it should be fine to pull in.

**Pull request merged**